ddtrace/internal/writer/writer.py

```diff
@@ -378,27 +379,41 @@ def flush_queue(self, raise_exc: bool = False):
     def _flush_queue_with_client(self, client: WriterClientBase, raise_exc: bool = False) -> None:
         n_traces = len(client.encoder)
         try:
-            encoded, n_traces = client.encoder.encode()
-
-            if encoded is None:
-                return
-
-            # Should gzip the payload if intake accepts it
-            if self._intake_accepts_gzip:
-                original_size = len(encoded)
-                # Replace the value to send with the gzipped the value
-                encoded = gzip.compress(encoded, compresslevel=6)
-                log.debug("Original size in bytes: %s, Compressed size: %s", original_size, len(encoded))
-
-                # And add the header
-                self._headers["Content-Encoding"] = "gzip"
-
+            payloads = client.encoder.encode()
+            if not isinstance(payloads, list):
+                n_traces = payloads[1]
+            elif len(payloads) == 1:
+                n_traces = payloads[0][1]
```

Comment:
```suggestion
            payloads = client.encoder.encode()
            if not isinstance(payloads, list):
                payloads = [payloads]
```

might be easier to manager, convert the old format to the list format ?

then we shouldn't have to check again in this file, we'll always coerce to a list, and can jut do `for encoded, n_traces in payloads: ...`